### PR TITLE
feat: Uplift Table for accessibility roles and custom attributes

### PR DESCRIPTION
**Pull request by okeeffed**
**Objective**

Ensure the table comes with base roles for ARIA and capability to accept HTML compliant props ie `aria-*` and `data-*`.

The compliant props follows on from work done previously for the `Paragraph` and `Header` which allows props accepted by `React.FunctionComponent` types.

**Notes**

There were no tests for Table in there and I am a little under the pump, so I placed in the bare minimum to validate the work I needed to do.

Happy enough for any touch ups but this build phase got me like 🙃 !

### CODE REVIEW

File: packages/component-library/draft/Kaizen/Table/Table.tsx [hunk 1]
```diff
@@ -8,18 +8,24 @@ const sortDescendingIcon = require("@kaizen/component-library/icons/sort-descend
...
+  <div className={styles.header} role="rowgroup" {...otherProps}>
+    {children}
+  </div>
 )
 
 type TableHeaderRow = React.FunctionComponent
-export const TableHeaderRow: TableHeaderRow = ({ children }) => (
-  <div className={styles.headerRow}>{children}</div>
+export const TableHeaderRow: TableHeaderRow = ({ children, ...otherProps }) => (
+  <div className={styles.headerRow} {...otherProps}>
```
Comment (didoesdigital) [inline packages/component-library/draft/Kaizen/Table/Table.tsx:26]: Without looking closely at the structure, does this need a ` role="rowheader"` or similar?

**Comment by didoesdigital** [comment]
Great work @okeeffed !

File: packages/component-library/draft/Kaizen/Table/Table.tsx (hunk 1 above)
Comment (okeeffed) [reply packages/component-library/draft/Kaizen/Table/Table.tsx:26]: Yep sure does! I will add that in.

**Comment by okeeffed** [comment]
@mbylstra I didn't know that either! I am glad you checked for my Friday TIL

**Pull request merged**